editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardTemplatePage.java

```diff
@@ -253,9 +254,46 @@ private void setTemplatesActive(boolean active) {
 			btn_blueprintDSL.setEnabled(getSelectedTemplate() != null && getSelectedTemplate().getTemplate().supportsDSL(CamelDSLType.BLUEPRINT));
 			btn_springDSL.setEnabled(getSelectedTemplate() != null && getSelectedTemplate().getTemplate().supportsDSL(CamelDSLType.SPRING));
 			btn_javaDSL.setEnabled(getSelectedTemplate() != null && getSelectedTemplate().getTemplate().supportsDSL(CamelDSLType.JAVA));
+			updateDSLButtonGroup(getSelectedTemplate());
 		}
 	}
 	
+	private void updateDSLButtonGroup(TemplateItem template) {
+		if (template == null || !disabledDSLSelected()) return;
+		for (CamelDSLType dslType : CamelDSLType.values()) {
```

Comment:
what do you think extracting a method selectFirstCompatibleDSLType(TemplateItem)?
it will help to understand the reason of this code.